dev/containers/barberChat.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { Component } from 'react';
+import io from 'socket.io-client';
+
+const socket = io('http://localhost:3000');
+
+// import { connect } from 'react-redux';
+
+// barber component joins room automatically when view is loaded
+// this way the user view can grab the corresponding barber email
+// when trying to chat and user view will emit a message to that room
+
+class BarberChat extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      term: '',
+      email: 'snypper@io.com',
+    };
+  }
+  componentDidMount() {
+    socket.emit('join', { email: 'snypper@io.com' });
+  }
+  handleChange(e) {
+    this.setState({ term: e.target.value });
+  }
+  handleSubmit() {
+    socket.emit('private-message', { email: this.state.email, msg: this.state.term });
+    this.setState({ term: '' });
+  }
+  render() {
+    return (
+      <form onSubmit={(e) => { e.preventDefault(); this.handleSubmit(this.state.term); }}>
```

Comment:
For both the onSubmit and onChange listeners, can you change it to just take the callback functions rather than invoking them in an anonymous function? It's a little hard to read and redundant as it is right now.  And remember to bind `this` context to the component's methods.